src/main/java/com/google/sps/data/Card.java

```diff
@@ -107,7 +119,57 @@ public Entity createEntity() {
     card.setProperty("imageBlobKey", this.imageBlobKey);
     card.setProperty("rawText", this.rawText);
     card.setProperty("textTranslated", this.textTranslated);
+    card.setProperty("status", ClassStatus.ALIVE.toString());
 
     return card;
   }
+
+  public static Card storeCardInDatastore(Card card, DatastoreService datastore, String folderKey) {
+    card.setParentKey(folderKey);
+    Entity cardEntity = card.createEntity();
+    datastore.put(cardEntity);
+
+    card.setCardKey(KeyFactory.keyToString(cardEntity.getKey()));
+
+    return card;
+  }
+
+  public static void deleteCard(Entity card) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    markCardAsDeleted(card);
+    try {
+      datastore.delete(card.getKey());
+    } catch (DatastoreFailureException e) {
+      throw e;
+    }
+  }
+
+  public static void deleteCardWithRetries(Entity card) {
+    int retries = 5;
```

Comment:
You could add this as a default argument.

Reply:
done